device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceTwin/DeviceTwin.java

```diff
@@ -281,17 +282,23 @@ public void getDeviceTwin()
         checkSubscription();
 
         IotHubTransportMessage getTwinRequestMessage = new IotHubTransportMessage(new byte[0], MessageType.DEVICE_TWIN);
-        getTwinRequestMessage.setRequestId(String.valueOf(requestId++));
+        getTwinRequestMessage.setRequestId(UUID.randomUUID().toString());
+        getTwinRequestMessage.setCorrelationId(getTwinRequestMessage.getRequestId());
```

Comment:
q - what is the difference in usage of request Id vs correlation Id?
Also, who sets `IotHubTransportMessage.setRequestId()`?

Reply:
This would only be set by the DeviceTwin methods that are aware of this interaction. So in this case the updateReportedProperties and getDeviceTwin(). However we don't expose any way to use the correlation context from the getDeviceTwin.

As for the use. 

When the message is sent over AMQP the correlation id is left intact as part of the message properties. But, when using MQTT properties are set via the topic. For generic event topics the property bag is left intact, but for our Twin topics only the resid is left intact and is the only way to correlate a request/response.

For AMQP the correlation id is left intact for the message request/response lifecycle.